staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -587,6 +587,9 @@ func (j JSONPathWait) checkCondition(obj *unstructured.Unstructured) (bool, erro
 	queryObj := obj.UnstructuredContent()
 	parseResults, err := j.jsonPathParser.FindResults(queryObj)
 	if err != nil {
+		if strings.Contains(err.Error(), "is not found") { // TODO(jonyhy96) change this to error assertion when client-go support.
+			return false, nil
+		}
```

Comment:
Rather than parsing the error string, did you consider using [AllowMissingKeys](https://github.com/kubernetes/client-go/blob/28ccde769fc5519dd84e5512ebf303ac86ef9d7c/util/jsonpath/jsonpath.go#L53-L55) and then when `len(parseResults) == 0` we can return `false, nil` [instead of an error](https://github.com/kubernetes/kubernetes/blob/a750d8054a6cb3167f495829ce3e77ab0ccca48e/staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go#L605-L607)?

Reply:
Looks great! Thanks for advice, then we can remove that todo.